### PR TITLE
(PXP-7097) Add optional alternate TopBar profile menu

### PR DESCRIPTION
**Pull request by olivarity**
Jira Ticket: [PXP-7097](https://ctds-planx.atlassian.net/browse/PXP-7097) 

### New Features
- Added new dropdown menu to `TopBar` component for handling user profile and logout options, opt-in via config `components.topBar.useProfileDropdown: true`

### Deployment changes

- At time of writing, `qa-midrc` is only env using the new feature flag. These changes should be deployed alongside each other to production, but nothing will break if done separately.

### CODE REVIEW

**Comment by mfshao** [comment]
is this thing (and the accompanying CSS change) deployed anywhere so we can take a look?

**Comment by olivarity** [comment]
> is this thing (and the accompanying CSS changes) deployed anywhere so we can take a look?

@mfshao waiting on uc-cdis/gitops-qa#1051 checks to finish, and then we'll deploy this version there :)

**Comment by mfshao** [comment]
> > is this thing (and the accompanying CSS changes) deployed anywhere so we can take a look?
> 
> @mfshao waiting on [uc-cdis/gitops-qa#1051](https://github.com/uc-cdis/gitops-qa/pull/1051) checks to finish, and then we'll deploy this version there :)

great! thanks

it would be easier if you put all these changes locally into your dev env so we can see it without waiting for jenkins

File: src/Layout/reduxer.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ export const ReduxTopBar = (() => {
   const mapStateToProps = state => ({
     navTitle: components.navigation.title,
     topItems: components.topBar.items,
+    useProfileDropdown: components.topBar.useProfileDropdown,
```
Comment (mfshao) [inline src/Layout/reduxer.js:44]: can you update `/docs/portal_config.md` about this new option as well?

File: src/components/layout/TopBar.jsx [hunk 2]
```diff
@@ -93,6 +94,32 @@ class TopBar extends Component {
...
+                  }
+                >
+                  <Link className='top-bar__link' to='#'>
+                    <TopIconButton
+                      icon='user-circle'
+                      name=''
+                      isActive={this.isActive('/identity')}
+                      onActiveTab={() => this.props.onActiveTab('/identity')}
+                    />
+                  </Link>
```
Comment (mfshao) [inline src/components/layout/TopBar.jsx:119]: I thought for this part there were some inconsistence in the design. Have we made a dicision that whether we display only a single icon or icon + username after user has logged in

If we decide to keep only a single icon, can you adjust the `placement` of the popover so the arrow gets pointed to the icon?

**Review by mfshao — COMMENTED** [review]
tested locally and lgtm in general with some minor comments

File: src/components/layout/TopBar.jsx (hunk 2 above)
Comment (olivarity) [reply src/components/layout/TopBar.jsx:119]: Either option is feasible, we'll have to put that one up to the team. Pushing a commit with correct `placement` now, that was just a temporary change that slipped through during experimentation.

File: src/Layout/reduxer.js (hunk 1 above)
Comment (olivarity) [reply src/Layout/reduxer.js:44]: Done

**Review by mfshao — APPROVED** [review]
lgtm 👍

**Pull request merged**